x-pack/spec/filters/geoip/database_manager_spec.rb

```diff
@@ -13,9 +14,10 @@
     let(:mock_download_manager)  { double("download_manager") }
     let(:mock_scheduler)  { double("scheduler") }
     let(:agent_metric)  { LogStash::Instrument::Metric.new(LogStash::Instrument::Collector.new) }
+    let(:database_metric) { LogStash::Filters::Geoip::DatabaseMetric.new(agent_metric) }
     let(:db_manager) do
       manager = Class.new(LogStash::Filters::Geoip::DatabaseManager).instance
-      manager.metric = agent_metric
+      manager.database_metric = database_metric
```

Comment:
Let's clean this up by resetting our state in an `after` hook, maybe by adding the following at line 40 of this file?

~~~ diff
diff --git a/x-pack/spec/filters/geoip/database_manager_spec.rb b/x-pack/spec/filters/geoip/database_manager_spec.rb
index f90faba24..a9dabcb16 100644
--- a/x-pack/spec/filters/geoip/database_manager_spec.rb
+++ b/x-pack/spec/filters/geoip/database_manager_spec.rb
@@ -37,6 +37,7 @@ describe LogStash::Filters::Geoip do
 
     after do
       delete_file(metadata_path, get_dir_path(second_dirname))
+      db_manager.database_metric = nil
     end
 
     context "initialize" do
~~~

Reply:
I am fine to set nil to be on the safe side